src/main/java/com/urbanairship/datacube/dbharnesses/HBaseDbHarness.java

```diff
@@ -326,39 +391,66 @@ private void flushBatch(Batch<T> batch) throws IOException, InterruptedException
 
         long nanoTimeBeforeBatch = System.nanoTime();
 
+
         try {
-            for (Map.Entry<Address, T> entry : batchMap.entrySet()) {
-                final Address address = entry.getKey();
-                final T op = entry.getValue();
-                final byte[] rowKey = ArrayUtils.addAll(uniqueCubeName, address.toWriteKey(idService));
-                final long nanoTimeBeforeWrite = System.nanoTime();
-
-                byte[] dbBytes;
-                switch (commitType) {
-                    case INCREMENT:
-                        long postIncr = increment(rowKey, op);
-                        dbBytes = Bytes.toBytes(postIncr);
-                        break;
-                    case READ_COMBINE_CAS:
-                        dbBytes = readCombineCas(rowKey, op);
-                        break;
-                    case OVERWRITE:
-                        overwrite(rowKey, op);
-                        dbBytes = op.serialize();
-                        break;
-                    default:
-                        throw new RuntimeException("Unsupported commit type " + commitType);
+            if (commitType.equals(CommitType.INCREMENT) && configuration.batchSize > 1) {
```

Comment:
Slight style preference for using the existing `switch` statement and putting branching logic that calls a method with the batching logic.

Reply:
We'd still need an extra branch outside of it, and the iteration happens completely differently. I just dont think that's worth risking the changes to the existing logic.

Reply:
Oh wait I totally missed that the `switch` statement is inside the loop going over the writes.